# MicroNexT
Microbial Next generation sequence analysis Tool
# ShotMAP

## Description
Microbial Next generation sequence analysis Tool (MicroNexT) is an easy-to-use, automated pipeline designed to analyze microbial genome sequencing data generated by next generation sequencing technology.
<p align="center">
<img src="https://github.com/ishtiaque-ahammad/ShotMAP/assets/99262870/c36acb52-0fac-46e1-aa19-c63f566017b4" alt="ShotMAP" />
</p>

## Features

* Quality control and trimming of sequencing reads (Fastp)
* De novo genome assembly (Unicycler)
* Genome annotation and gene prediction (Prokka)
* Metabolic pathway analysis (Prodigal, MicrobeAnnotator)
* Identification of antibiotic resistance genes (Abricate)
* Secondary metabolite identification (Antismash)
Identification of CRISPR-Cas systems (Cctyper)

## Dependencies
ShotMAP relies on several software tools and requires them to be installed. These tools include:

* [Fastp](https://github.com/OpenGene/fastp)

* [Unicycler](https://github.com/rrwick/Unicycler)

* [BBMap](https://github.com/BioInfoTools/BBMap)
 
* [Prokka](https://github.com/tseemann/prokka)

* [Prodigal](https://github.com/hyattpd/Prodigal)

* [MicrobeAnnotator](https://github.com/cruizperez/MicrobeAnnotator)

* [Abricate](https://github.com/tseemann/abricate)

* [Antismash](https://github.com/antismash)
  
* [Cctyper](https://github.com/Russel88/CRISPRCasTyper)

## Installation

### Clone the repository
``` git clone https://github.com/ishtiaque-ahammad/ShotMAP ```

### Navigate to the cloned directory
``` cd ShotMAP ```

### Make the script executable
``` chmod +x ShotMAP.sh ```

### Install dependencies
``` conda env create --file fastp.yml ```

``` conda env create --file megahit.yml ```

``` conda env create --file metabat2.yml ```

``` conda env create --file checkm.yml ```

``` conda env create --file prodigal.yml ```

``` conda env create --file abricate.yml ```

``` conda env create --file dbcan.yml ```

``` conda env create --file microbeannotator.yml ```

## Usage

### Prepare your data

Ensure you have your forward and reverse read files in fastq / fastq.gz format with their extensions (e.g., R1.fastq.gz, R2.fastq.gz).

### Run the script
```  bash ShotMAP.sh ``` 

### Output
The script will create separate directories for each analysis step and store the corresponding output files.

## License
This project is licensed under the GPL-3.0 license.
