# MicroNexT

## Description
Microbial Next generation sequence analysis Tool (MicroNexT) is an easy-to-use, automated pipeline designed to analyze microbial genome sequencing data generated by next generation sequencing technology.
<p align="center">
<img src="https://github.com/ishtiaque-ahammad/ShotMAP/assets/99262870/c36acb52-0fac-46e1-aa19-c63f566017b4" alt="ShotMAP" />
</p>

## Features

* Quality control and trimming of sequencing reads (Fastp)
* De novo genome assembly (Unicycler)
* Genome annotation and gene prediction (Prokka)
* Metabolic pathway analysis (Prodigal, MicrobeAnnotator)
* Identification of antibiotic resistance genes (Abricate)
* Secondary metabolite identification (Antismash)
* Identification of CRISPR-Cas systems (Cctyper)

## Dependencies

* [Fastp](https://github.com/OpenGene/fastp)

* [Unicycler](https://github.com/rrwick/Unicycler)

* [BBMap](https://github.com/BioInfoTools/BBMap)
 
* [Prokka](https://github.com/tseemann/prokka)

* [Prodigal](https://github.com/hyattpd/Prodigal)

* [MicrobeAnnotator](https://github.com/cruizperez/MicrobeAnnotator)

* [Abricate](https://github.com/tseemann/abricate)

* [Antismash](https://github.com/antismash)
  
* [Cctyper](https://github.com/Russel88/CRISPRCasTyper)

## Installation

### Clone the repository
``` git clone https://github.com/ishtiaque-ahammad/MicroNexT ```

### Navigate to the cloned directory
``` cd MicroNexT ```

### Make the script executable
``` chmod +x micronext.sh ```

### Install dependencies

``` conda env create --file fastp.yml ```

``` conda env create --file unicycler.yml ```

``` conda env create --file bbmap.yml ```

``` conda env create --file prokka.yml ```

``` conda env create --file prodigal.yml ```

``` conda env create --file microbeannotator.yml ```

``` conda env create --file abricate.yml ```

``` conda env create --file antismash.yml ```

``` conda env create --file cctyper.yml ```

## Usage

### Run the script

```  bash micronext.sh ``` 

The script will prompt you for the following information:

* Forward read file name
* Reverse read file name
* Genus name
* Species name
* Number of CPU cores

The script will then proceed with the analysis steps and generate outputs in designated folders.

### Output
MicroNexT organizes the outputs of various analysis in different folders within the directory you run the script from. These include:

1. major_output_files: Contains core outputs like assembled genome, annotated genome files, and summary tables.

2. assembled_genome: Stores files related to genome assembly.

3. annotated_genome: Contains outputs from Prokka annotation.

4. metabolic_pathways: Stores results from MicrobeAnnotator pathway analysis.

5. antibiotic_resistant_genes: Contains identified antibiotic resistance genes.

6. biosynthetic_gene_clusters: Stores outputs from Antismash related to secondary metabolite identification.

7. crispr_sites: Contains results from Cctyper regarding CRISPR-Cas systems.

## License
This project is licensed under the GPL-3.0 license.
